include/jemalloc/internal/prof_inlines_a.h

```diff
@@ -3,11 +3,36 @@
 
 #include "jemalloc/internal/mutex.h"
 
+JEMALLOC_ALWAYS_INLINE void
+prof_active_assert() {
+	cassert(config_prof);
+	/*
+	 * If opt_prof is off, then prof_active must always be off, regardless
+	 * of whether prof_active_mtx is in effect or not.
+	 */
+	assert(opt_prof || !prof_active);
+}
+
+JEMALLOC_ALWAYS_INLINE bool
+prof_active_get_unlocked(void) {
+	prof_active_assert();
+	/*
+	 * Even if opt_prof is true, sampling can be temporarily disabled by
+	 * setting prof_active to false.  No locking is used when reading
+	 * prof_active in the fast path, so there are no guarantees regarding
+	 * how long it will take for all threads to notice state changes.
+	 */
+	return prof_active;
+}
+
 static inline bool
-prof_accum_add(tsdn_t *tsdn, prof_accum_t *prof_accum,
-    uint64_t accumbytes) {
+prof_idump_accum(tsdn_t *tsdn, uint64_t accumbytes) {
 	cassert(config_prof);
 
+	if (prof_interval == 0 || !prof_active_get_unlocked()) {
+		return false;
+	}
+
```

Comment:
nit: move the stuff below to a static function, maybe prof_idump_accum_impl?

Reply:
You mean a non-static and non-inlined function in `prof.c`? How about putting the entire function there as well? For example, functions like `prof_idump()` are also there and have similar early termination conditions.

Reply:
Yeah I meant a non-static function in prof.c. None of this happens often, so this shouldn't matter. I slightly prefer keeping the above branch inlined, but the body blow non inlined.